src/tickit/core/components/device_simulation.py

```diff
@@ -74,3 +80,10 @@ async def on_tick(self, time: SimTime, changes: Changes) -> None:
         )
         self.last_outputs = device_update.outputs
         await self.output(time, out_changes, device_update.call_at)
+
+    async def stop_component(self) -> None:
+        """Cancel pending tasks associated with the component."""
+        LOGGER.debug("Stopping {}".format(self.name))
+        for task in self._tasks:
+            task.cancel()
+            await task
```

Comment:
You can improve efficiency with the following, which ensures that all tasks start the process of cancelling before you block on any one of them
```suggestion
        for task in self._tasks:
            task.cancel()
        await asyncio.wait(tasks)
```